views/app.register.fg

```diff
@@ -77,13 +77,22 @@ POST odin/app/register / {
 
 # Existing username register twice should failed
 # TODO: Now it returns std::exception
-# POST odin/app/register / {
-#     "username": "fred",
-#     "password": "mercury2",
-#     "password2": "mercury2",
-#     "full-name": "Barney",
-#     "email": "fred2@bowling.com"
-# } 201
+sql.insert odin.identity {"id": "installation_ref2"}
+sql.insert odin.app_user_installation_id_ledger {
+    "reference": "reference2",
+    "app_id": "bowling-app",
+    "identity_id": "installation_ref2",
+    "installation_id": "fred-mobile2"
+}
+set app_jwt (odin.jwt.mint {"sub": "installation_ref2", "iss": "http://odin.felspar.com/app/bowling-app"} <JWT_SECRET>bowling-app)
+set-path testserver.headers ["Authorization"] (cat "Bearer " (lookup app_jwt))
+POST odin/app/register / {
```

Comment:
@KayEss This test doesn't work because it throw `std::exception`.
Test log
```
Fostgres testing environment
Going to be using database odin-test-view-app.register
Database found. Dropping odin-test-view-app.register
Creating database odin-test-view-app.register
Loading library /home/tle/proteus/KayEss/fost-dev/fost-meta/build.tmp/clang--release/odin/Cpp/odin-fg/libodin-fg.so
Loading library /home/tle/proteus/KayEss/fost-dev/fost-meta/build.tmp/clang--release/odin/Cpp/odin-views/libodin-views.so
Executing SQL /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/../Schema/bootstrap.sql
Loading configuration /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/../Configuration/odin-views.json
Loading configuration /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/../Configuration/odin-federation-server.json
Loading configuration /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/app.register.json
Loading configuration /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/app.register.test.json
Loading script /home/tle/proteus/KayEss/fost-dev/fost-meta/odin/views/app.register.fg
Caught std::exception

ERROR:  duplicate key value violates unique constraint "login_uq"
DETAIL:  Key (login)=(fred) already exists.
CONTEXT:  SQL statement "INSERT
            INTO odin.credentials
                (identity_id, login, password__hash, password__process, password__reference)
            VALUES (NEW.identity_id, NEW.login, NEW.password, NEW.process, NEW.reference)
            ON CONFLICT (identity_id) DO UPDATE SET
                password__hash = EXCLUDED.password__hash,
                password__process = EXCLUDED.password__process,
                password__reference = EXCLUDED.password__reference"
PL/pgSQL function credentials_ledger_insert() line 3 at SQL statement
```
Any suggestion?

Reply:
You need to catch the exception in the view and decide what to turn it into. This is where we could really do with the Postgres exception wrapper view that can handle the exceptions and forward on to specific views.

Reply:
Ok, so I will add try-catch exception to `fostgres.sql` view